woodwork/type_sys/type_system.py

```diff
@@ -252,16 +254,25 @@ def infer_logical_type(self, series):
         Args:
             series (pandas.Series): The series for which to infer the LogicalType.
         """
+        if isinstance(series, pd.Series):
+            # Special case for series with no valid values
+            if series.count() == 0:
+                return Unknown()
 
-        # Bring Dask or Koalas data into memory for inference
-        if _is_dask_series(series):
-            series = series.get_partition(0).compute()
-        if _is_koalas_series(series):
-            series = series.head(100000).to_pandas()
+            series = series.head(100000)
+        else:
+            if _is_dask_series(series):
+                series = series.head(100000)
+            elif _is_koalas_series(series):
+                series = series.head(100000).to_pandas()
+            else:
+                raise ValueError(f"Unexpected arg type `{type(series)}`")  # pragma: no cover
```

Comment:
I'm not sure we need to include this else clause, since we don't claim to support any other dataframe/series types beyond pandas, Dask or Koalas.

Reply:
For user interpretability, would  "Unexpected series type" or "Unsupported series type"

Reply:
Sure, I can make that change.

Reply:
@davesque Makes sense, and I can see the benefit of leaving this if we do add support for new types, this could help alert us to a new condition we need to handle.